tests/drawer/test_draw.py

```diff
@@ -300,6 +300,32 @@ def circ():
     spy.assert_called_once()
 
 
+@pytest.mark.parametrize(
+    "postselect, reset, mid_measure_label",
+    [
+        (None, False, "┤↗├"),
+        (None, True, "┤↗│  │0⟩"),
+        (0, False, "┤↗₀├"),
+        (0, True, "┤↗₀│  │0⟩"),
+        (1, False, "┤↗₁├"),
+        (1, True, "┤↗₁│  │0⟩"),
+    ],
+)
+def test_draw_mid_circuit_measurement(postselect, reset, mid_measure_label):
+    """Test that mid-circuit measurements are drawn correctly."""
+
+    def func():
+        qml.Hadamard(0)
+        qml.measure(0, reset=reset, postselect=postselect)
+        qml.PauliX(0)
+        return qml.expval(qml.PauliZ(0))
+
+    drawing = qml.draw(func)()
+    expected_drawing = "0: ──H──" + mid_measure_label + "──X─┤  <Z>"
+
+    assert drawing == expected_drawing
+
+
```

Comment:
@mudit2812 is it worth having an integration-level test? E.g., creating the circuit we've shared in Slack and checking if it matches the expected text-based output?

Reply:
In my mind, this is an integration level test.

Reply:
Yep sorry this test is probably fine, I'm just curious if we could test for the circuit we've discussed in Slack because that's the one that we've all agreed upon.

Reply:
Thanks!